docs/design/resource-states.md

```diff
@@ -0,0 +1,222 @@
+
+Some other best practices on [Kubernetes design principles](https://github.com/kubernetes/community/blob/4c9ef2d/contributors/design-proposals/architecture/principles.md):
+
+> Object status must be 100% reconstructable by observation. Any history kept must be just an optimization and not required for correct operation.
+
+> Do not define comprehensive state machines for objects with behaviors associated with state transitions and/or "assumed" states that cannot be ascertained by observation.
+
+### Details
+All resource's `Status`'s will have a top level `Conditions` field which is a collection of type `Condition`. This collection supports extension (through conditions with different type names).
+Initially we will expose a single `Ready` condition across all resources, representing the high level availability of the resource and its readiness for use.
+
+The structure of the `Condition` is based on the [reccomended shape of conditions KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1623-standardize-conditions) as well as the [Cluster API conditions propsal](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/20200506-conditions.md) and is as follows:
+
+```go
+// ConditionSeverity expresses the severity of a Condition Type failing.
+type ConditionSeverity string
+
+const (
+  // ConditionSeverityError specifies that a failure of a condition type
+  // should be viewed as an error.
+  ConditionSeverityError ConditionSeverity = "Error"
+
+  // ConditionSeverityWarning specifies that a failure of a condition type
+  // should be viewed as a warning, but that things could still work.
+  ConditionSeverityWarning ConditionSeverity = "Warning"
+
+  // ConditionSeverityInfo specifies that a failure of a condition type
+  // should be viewed as purely informational, and that things could still work.
+  ConditionSeverityInfo ConditionSeverity = "Info"
+
+  // ConditionSeverityNone will only be set if the condition is in state "True".
+  // This is the default state for conditions.
+  ConditionSeverityNone ConditionSeverity = ""
+)
+
+// Condition defines an extension to status (an observation) of an ASO resource
+type Condition struct {
+   // Type of condition.
+   // +required
+   Type string `json:"type"`
+
+   // Status of the condition, one of True, False, or Unknown.
+   // +required
+   Status corev1.ConditionStatus `json:"status"`
+
+   // Severity with which to treat failures of this type of condition.
+   // For conditions which have positive polarity (Status == True is their normal/healthy state), this will be omitted when Status == True
+   // For conditions which have negative polarity (Status == False is their normal/healthy state), this will be omitted when Status == False.
+   // +optional
+   Severity ConditionSeverity `json:"severity,omitempty"`
+
+   // LastTransitionTime is the last time the condition transitioned from one status to another.
```

Comment:
```suggestion
   // LastTransitionTime is the last time the condition changed
```

Reply:
See my comment here: https://github.com/Azure/azure-service-operator/pull/1614#issuecomment-875930699

I've added this as an open question and we should decide which we prefer. CAPI says it just changes when `Status` changes, but I checked their actual implementation and it really changes anytime the condition changes at all (even if it's from status False -> status False.

The KEP for conditions says:
> 	// Last time the condition transitioned from one status to another.
	// This should be when the underlying condition changed.  If that is not known, then using the time when the API field changed is acceptable

Which makes me think we should make it as you comment here...

Reply:
I've updated to add a new section called Open questions (answered), where I posed and answered this question. Have updated the comment to match what you suggested.